component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/voiceotp/VoiceOTPAuthenticator.java

```diff
@@ -0,0 +1,2037 @@
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.wso2.carbon.identity.authenticator.voiceotp.VoiceOTPConstants.CHAR_SET_UTF_8;
+import static org.wso2.carbon.identity.authenticator.voiceotp.VoiceOTPConstants.CONTENT_TYPE;
+import static org.wso2.carbon.identity.authenticator.voiceotp.VoiceOTPConstants.ERROR_MESSAGE_DETAILS;
+import static org.wso2.carbon.identity.authenticator.voiceotp.VoiceOTPConstants.JSON_CONTENT_TYPE;
+import static org.wso2.carbon.identity.authenticator.voiceotp.VoiceOTPConstants.MASKING_VALUE_SEPARATOR;
+import static org.wso2.carbon.identity.authenticator.voiceotp.VoiceOTPConstants.MOBILE_NUMBER_REGEX;
+import static org.wso2.carbon.identity.authenticator.voiceotp.VoiceOTPConstants.POST_METHOD;
+import static org.wso2.carbon.identity.authenticator.voiceotp.VoiceOTPConstants.REQUESTED_USER_MOBILE;
+import static org.wso2.carbon.identity.authenticator.voiceotp.VoiceOTPConstants.RESEND;
+import static org.wso2.carbon.identity.authenticator.voiceotp.VoiceOTPConstants.XML_CONTENT_TYPE;
+
+import static java.util.Base64.getEncoder;
+
+/**
+ * Authenticator of Voice OTP.
+ */
+public class VoiceOTPAuthenticator extends AbstractApplicationAuthenticator
+        implements FederatedApplicationAuthenticator {
+
+    private static final Log log = LogFactory.getLog(VoiceOTPAuthenticator.class);
+
+    /**
+     * Check whether the authentication or logout request can be handled by the authenticator.
+     *
+     * @param request The HttpServletRequest
+     * @return whether the authenticator can handle the inbound request
+     */
+    @Override
+    public boolean canHandle(HttpServletRequest request) {
+
+        log.debug("Inside VoiceOTPAuthenticator canHandle method and check the existence of mobile number and " +
+                    "otp code");
+
+        return ((StringUtils.isNotEmpty(request.getParameter(VoiceOTPConstants.RESEND))
+                && StringUtils.isEmpty(request.getParameter(VoiceOTPConstants.CODE)))
+                || StringUtils.isNotEmpty(request.getParameter(VoiceOTPConstants.CODE))
+                || StringUtils.isNotEmpty(request.getParameter(VoiceOTPConstants.MOBILE_NUMBER)));
+    }
+
+    /**
+     * Check whether the authentication or logout request can be handled by the authenticator.
+     *
+     * @param request The HttpServletRequest
+     * @param response The HttpServletResponse
+     * @param context The AuthenticationContext
+     * @return AuthenticatorFlowStatus type
```

Comment:
Add a more meaningful method comment. The description does not seem to be reflecting the implementation of this method.